src/api/app/components/build_status_badge_component.rb

```diff
@@ -0,0 +1,36 @@
+class BuildStatusBadgeComponent < ApplicationComponent
+  def initialize(status:, text:, url: nil)
+    super
+
+    @status = status
+    @text = text
+    @url = url
+    @category = Buildresult::BUILD_STATUS_CATEGORIES_MAP[status]
+  end
+
+  ICON = {
+    succeeded: 'fa-check',
+    failed: 'fa-circle-exclamation',
+    unresolvable: 'fa-circle-exclamation',
+    broken: 'fa-circle-exclamation',
+    blocked: 'fa-shield',
+    scheduled: 'fa-hourglass-half',
+    dispatching: 'fa-plane-departure',
+    building: 'fa-gear',
+    signing: 'fa-signature',
+    finished: 'fa-check',
+    disabled: 'fa-ban',
+    excluded: 'fa-ban',
+    locked: 'fa-lock',
+    deleting: 'fa-eraser',
+    unknown: 'fa-question'
+  }.with_indifferent_access.freeze
+
+  def badge_color
+    BuildStatusCountBadgeComponent::CATEGORY_BADGE_COLOR[@category]
+  end
+
+  def icon
```

Comment:
I think `icon` should go also into the view, it belongs there and there is no logic to extract